Library/Homebrew/github_packages.rb

```diff
@@ -175,54 +171,28 @@ def upload_bottle(user, token, skopeo, bottle_hash, dry_run:)
 
       tar_gz_sha256 = write_tar_gz(local_file, blobs)
 
-      tab = Tab.from_file_content(
-        Utils.safe_popen_read("tar", "xfO", local_file, "#{formula_name}/#{version}/INSTALL_RECEIPT.json"),
-        "#{local_file}/#{formula_name}/#{version}",
-      )
-      os_version = if tab.built_on.present?
-        /(\d+\.)*\d+/ =~ tab.built_on["os_version"]
-        Regexp.last_match(0)
-      end
-
-      # TODO: ideally most/all of these attributes would be stored in the
-      # bottle JSON rather than reading them from the formula.
-      os, arch, formulae_dir = if bottle_tag.to_s.end_with?("_linux")
-        ["linux", "amd64", "formula-linux"]
-      else
-        os = "darwin"
-        macos_version = MacOS::Version.from_symbol(bottle_tag.to_sym)
-        os_version ||= macos_version.to_f.to_s
-        arch = if macos_version.arch == :arm64
-          "arm64"
-        else
-          "amd64"
-        end
-        [os, arch, "formula"]
-      end
-
+      tab = bottle_tag["tab"]
       platform_hash = {
-        architecture: arch,
-        os: os,
-        "os.version" => os_version,
+        architecture: tab["built_on"]["arch"],
+        os: tab["built_on"]["os"],
+        "os.version" => tab["built_on"]["os_version"],
```

Comment:
On Linux, using the distribution version without qualifying it with the distribution name is seems ambiguous. I'd suggest either
```json
{ "os": "linux", "os.version": "Ubuntu 16.04" }
```
or instead storing the glibc version:
```json
{ "os": "linux", "os.version": "glibc 2.23" }
```

Reply:
I believe this does already set the distribution name as it's calling `lsb_release`. If not: this should also be fixed in the tab.